internal/graph/preloadedtaskrunner.go

```diff
@@ -84,8 +99,9 @@ func (tr *preloadedTaskRunner) runner() {
 			// Run the task. If an error occurs, store it and cancel any further tasks.
 			err := task(tr.ctx)
 			if err != nil {
-				tr.cancel()
+				tr.storeErrorAndCancel(err)
 			}
+			tr.wg.Done()
```

Comment:
can't `startAndWait` deadlock if the task errors and we don't call done here?

Reply:
the conditional falls through on error